cs-verifiable-encryption/README.md

```diff
@@ -0,0 +1,136 @@
+**Note**: The *prover* should never encrypt his link secret for the auditor.
+The verifiable encryption follows the Sigma protocol approach where the prover first creates the ciphertext of the attributes and ciphertext of blindings for those attributes. These blindings should match the ones used in the overall proof. After generating/receiving the challenge, it computes the response. The verifier will then use challnge and response to reconstruct the ciphertext over blindings and check for equality. The reason to structure it this way is for composability such that the main proving protocol should be able to use it.  
+This verifiable encryption is meant to be treated as a predicate.  
+The next section will describe the API of this protocol.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This protocol has been implemented in this [PR](https://github.com/hyperledger/ursa/pull/40). The PR contains tests to demonstrate how the protocol works.
+
+1. The auditor can create his encryption keypair using `CSKeypair::new`. This method takes the maximum number of attributes that can be encrypted using the keypair. The following example creates a keypair whose public key can be used to encrypt at max 3 attributes. Trying to encrypt more will result in an error.  The `CSKeypair` is an struct of 2 fields, a public key `CSEncPubkey` and a private key `CSEncPrikey`.  
+
+    ```rust
+    let keypair: CSKeypair = CSKeypair::new(3)?;
+    let pub_key: &CSEncPubkey = &keypair.pub_key;
+    let pri_key: &CSEncPrikey = &keypair.pri_key;
+    ```
+
+1. Say the prover now wants to encrypt 3 attributes. He creates a sequence of his 2 attributes uses the above keypair's public key to call `CSKeypair::encrypt`. The `CSKeypair::encrypt` in addition to the public key takes a `label` which is a public value and indicates the condition underwhich the ciphertext should be decrypted. The same `label` should be used while decrypting this ciphetext. The `CSKeypair::encrypt` function returns a `CSCiphertext` if there is no error. 
+    
+    ```rust
+    let pub_key: &CSEncPubkey = &keypair.pub_key;
```

Comment:
Probably better to show the prover deserializing the published public key here. In practice the prover would not be referencing a CSKeypair object.

Reply:
Done.